test/mitmproxy/addons/test_maplocal.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+
+from mitmproxy.addons.maplocal import MapLocal, file_candidates
+from mitmproxy.test import taddons
+from mitmproxy.test import tflow
+
+from mitmproxy.addons.modifyheaders import parse_modify_spec
+
+
+@pytest.mark.parametrize(
+    "url,spec,expected_candidates", [
+        (
+            "https://example.org/img/topic/subtopic/test.jpg",
+            ":example.com/foo:/tmp",
+            ["/tmp/img/topic/subtopic/test.jpg", "/tmp/img/topic/test.jpg", "/tmp/img/test.jpg", "/tmp/test.jpg"]
+        ),
+        (
+            "https://example.org/img/topic/subtopic/",
+            ":/img:/tmp",
+            ["/tmp/img/topic/subtopic/index.html", "/tmp/img/topic/index.html", "/tmp/img/index.html", "/tmp/index.html"]
+        ),
+        (
+            "https://example.org",
+            ":org:/tmp",
+            ["/tmp/index.html"]
+        ),
+    ]
```

Comment:
We should have a better test coverage here. How about varying prefix subjects (`example.com`, `example.com/foo`, `example.com/foo/`) and URLs including query strings, for example?